java/system-test/src/main/java/sleeper/systemtest/cdk/SystemTestApp.java

```
@@ -75,7 +76,11 @@ protected void generateProperties() {
 
     @Override
     protected SystemTestProperties getInstanceProperties() {
-        return (SystemTestProperties) super.getInstanceProperties();
+        InstanceProperties properties = super.getInstanceProperties();
+        if (properties instanceof SystemTestProperties) {
+            return (SystemTestProperties) properties;
+        }
+        return null;
```

I think we'd rather throw an exception than return null here.